### PR TITLE
[FEATURE] Implements max_doc_length support

### PR DESCRIPTION
**Pull request by joelschutz**
Implements support for differentiation between `max-doc-length` and `max-line-length`.

### CODE REVIEW

**Comment by orsinium** [comment]
Cool. Could you tell a bit more about the motivation behind, please?

**Comment by joelschutz** [comment]
> Cool. Could you tell a bit more about the motivation behind, please?

Our company style-guide defines different line length limits for code and comment lines. As we were using the plugin we noticed that it doesn't make that distinction, so I made a quick implementation of that feature for us to use.

Later today I will write the PR description in more detail. Feel free to indicate changes and corrections as you see fit.

**Comment by orsinium** [comment]
Nah, it's fine, go on. I'm just wondering what might be the motivation to have a different line length limit for docstrings. Like, the main motivation to have a limit is to make it readable and fit on most of the screen, and it should be about the same for docstrings, code, comments, etc.

File: pyproject.toml [hunk 1]
```diff
@@ -49,3 +49,4 @@ dev = [
 
 [tool.flit.entrypoints."flake8.extension"]
 LN001 = "flake8_length:Checker"
+LN002 = "flake8_length:Checker"
```
Comment (orsinium) [inline pyproject.toml]: let's have one entrypoint, just a bit broader:

```suggestion
LN00 = "flake8_length:Checker"
```

File: flake8_length/_checker.py [hunk 2]
```diff
@@ -1,38 +1,46 @@
...
+VIOLATION_DESCRIPTION_MAP = {
+    'LN001': 'code line is too long',
+    'LN002': 'doc/comment line is too long'
+}
+
+TEMPLATE = '{v.code} {d} ({v.length} > {v.limit})'
 
 
 class Violation(NamedTuple):
+    code: Literal['LN001', 'LN002']
```
Comment (orsinium) [inline flake8_length/_checker.py]: `Literal` is new in 3.8, the current stable is 3.7, let's just use `str` there.

```suggestion
    code: str
```

**Comment by joelschutz** [comment]
> Nah, it's fine, go on. I'm just wondering what might be the motivation to have a different line length limit for docstrings. Like, the main motivation to have a limit is to make it readable and fit on most of the screen, and it should be about the same for docstrings, code, comments, etc.

We decided to use this feature to make our dostring blocks smaller and less intrusive. Since it doesn't make sense for all cases to have different limit, I assumed the max_line_length for both if no max_doc_length is provided.

**Comment by orsinium** [comment]
Thank you! I was a bit in doubt that we now consider comments as docs but we can improve it later. I think that having a separate limit for docs is a rare case anyway.

**Pull request merged**